sound/soc/sof/topology.c

```diff
@@ -809,6 +809,68 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,  void *object,
 				    array_size, 1, 0);
 }
 
+static int sof_w_no_wname_in_long_name(struct snd_soc_component *scomp,
+				       struct snd_soc_dapm_widget *w, int type,
+				       struct snd_soc_tplg_vendor_value_elem *v)
+{
+	if (!(type == SND_SOC_TPLG_TUPLE_TYPE_WORD ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_SHORT ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_BYTE ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_BOOL)) {
+		dev_err(scomp->dev, "Bad tuple type %d\n", type);
+		return -EINVAL;
```

Comment:
I don't understand what this error check does?

Reply:
If the tuple data type is not something that can easily be interpreted as boolean, then there is reason to believe that the topology is otherwise broken too. But sure, I can change this to a warning only.